### PR TITLE
Fix for #742

### PR DESCRIPTION
**Pull request by dkd903**

### CODE REVIEW

File: app/Http/Controllers/UserManageApiController.php [hunk 1]
```diff
@@ -115,15 +115,33 @@ public function postSignup()
...
+                    $message->to($email); // Recipient address
+                });
+
+                return Response::json(array( 'status' => 'ok', 'errors' => array(), 'message' => 'An email has been sent to your email address.  Please follow the instructions in the email to confirm your email address before logging in.'));
+
+            } catch (\Exception $e) {
+
+                //auto verify the user and log them in
+                //in case sending the email fails
+                $user = User::where('id', $user_id)->first();
```
Comment (krues8dr) [inline app/Http/Controllers/UserManageApiController.php]: @dkd903 You shouldn't need to re-find the user here, you've still got a reference to `$user` from above which is valid in this scope.

**Comment by krues8dr** [comment]
@dkd903 It looks like you're doing all of your work out of your `master` branch, so *all* of your commits are getting squeezed into your pull requests.  We use [git flow](http://nvie.com/posts/a-successful-git-branching-model/) internally to avoid this, by creating a new branch for each issue that we're resolving, e.g. `issue-731` etc.  It looks like you've done a lot of work here, so we can handle what's here to save you some trouble splitting all of this up.

**Comment by dkd903** [comment]
Sorry about that @krues8dr and thank you. I will create new branches corresponding to the issue# going forward and then send the PRs.

File: app/Models/Annotation.php [hunk 2]
```diff
@@ -313,6 +313,10 @@ public function toAnnotatorArray($userId = null)
         $item['flags'] = $this->flags();
         $item['seen'] = $this->seen;
 
+        //fix for #804: Time stamps for annotations are off by several hours
+        //convert date to RFC2822 string
+        $item['created_at'] = date_format(date_create($item['created_at']), "r");
+
```
Comment (krues8dr) [inline app/Models/Annotation.php]: @dkd903 This is great, but you'll need to do this for both `created_at` and `updated_at` for consistency across the site.  Were you able to determine why `toRFC2822String()` above was mangling the timezone info?

File: app/Models/AnnotationComment.php [hunk 3]
```diff
@@ -25,18 +25,30 @@ public function user()
...
+    *   reformat the date structure for the datastore #804.
+    *
+    * @param string $date
+    *
+    * @return Carbon date
+    */
+    public function getCreatedAtAttribute($date)
+    {
+        return \Carbon\Carbon::createFromFormat('Y-m-d H:i:s', $date)->format('r');
+    }
```
Comment (krues8dr) [inline app/Models/AnnotationComment.php]: @dkd903 Also here, you'll need to add handling for the `updated_at` field.

**Comment by krues8dr** [comment]
This looks good except for the few comments I've added!  Resolve those and we'll be good to go!  Thanks!

**Comment by dkd903** [comment]
@krues8dr have pushed the changes from the Code Review, please take a look.

**Comment by krues8dr** [comment]
This looks fantastic.  Excellent work @dkd903 !

**Comment by krues8dr** [comment]
Merged to `dev` branch.  Thanks for the great work on this @dkd903 !

**Pull request closed without merging**